### ISSUE TITLE
Add Neural Network Layer functions to Mindspore frontend

### ISSUE THREAD

**Issue by Ishticode** [unifyai/ivy#14951]
These functions can be found in [`mindspore.ops`](https://jax.readthedocs.io/en/latest/jax.numpy.html) namespace.

_

>Please keep in mind that the proper way to link an issue to this list is to comment "- [ ] #issue_number" while the issue's title only includes the name of the function you've chosen.

- [ ] adaptive_avg_pool2d
- [ ] adaptive_avg_pool3d
- [ ] adaptive_max_pool2d
- [ ] avg_pool2d
- [ ] bias_add
- [ ] conv2d
- [ ] ctc_greedy_decoder
- [ ] deformable_conv2d
- [ ] dropout2d
- [ ] dropout3d
- [ ] flatten
- [ ] interpolate
- [ ] lrn
- [ ] max_pool3d
- [ ] kl_div
- [ ] pad
- [ ] padding

**Comment by championii** [comment]
-[ ] #15002

**Comment by championii** [comment]
- [ ] #15002

**Comment by championii** [comment]
- [ ] #15002

**Comment by Unicorncosmos** [comment]
- [ ] #16674

**Comment by Unicorncosmos** [comment]
- [ ] #16674

**Comment by vmurari17** [comment]
- [ ]  softsign #17758

**Comment by vmurari17** [comment]
- [ ] #17768

**Comment by rishabh0098** [comment]
- [] #17800

**Comment by rishabh0098** [comment]
- [ ] #17818

**Comment by diksha-shrivastava13** [comment]
- [ ] #18209

**Comment by vmurari17** [comment]
@Ishticode could you mark softsign as done since I my PR got merged

**Comment by Ishticode** [comment]
> vmurari17
@vmurari17 
sure I can do that. pls link the pr for me and the issue if it was opened. If not I can create one. Thanks

**Comment by ChrisEduS** [comment]
#19661

**Comment by hedie-rgb** [comment]
- sigmoid #20118

**Comment by themaigod** [comment]
<conv2d> for Neural Network Layer functions to Mindspore frontend #20120

**Comment by Zamokuhle23** [comment]
adaptive_avg_pool2d #20804

**Comment by elabongaatuo** [comment]
- [ ] #20863

**Comment by elabongaatuo** [comment]
- [ ] #20864

**Comment by themaigod** [comment]
- [conv2d for Neural Network Layer functions to Mindspore frontend] #20120

**Comment by themaigod** [comment]
- [conv2d] #20120

**Comment by themaigod** [comment]
- [deformable_conv2d] #20945

**Comment by themaigod** [comment]
- [conv2d] #20120

**Comment by themaigod** [comment]
- [deformable_conv2d] #20945

**Comment by themaigod** [comment]
- [conv1d] #20961

**Comment by themaigod** [comment]
- [conv3d] #20962

**Comment by sushmanthreddy** [comment]
@Ishticode can u pls review the above pull request?

**Comment by sushmanthreddy** [comment]
@Ishticode pls review the dropout 3d also

**Comment by sushmanthreddy** [comment]
[pad] #21044

**Comment by sushmanthreddy** [comment]
[pad] #21044

**Comment by sushmanthreddy** [comment]
- [gumbel_softmax] #21043

**Comment by sushmanthreddy** [comment]
- [pad] #21044

**Comment by sushmanthreddy** [comment]
- [kl_div] #21045

**Comment by sushmanthreddy** [comment]
- [adaptive_avg_pool3d] #21046

**Comment by sushmanthreddy** [comment]
- [lrm] #21047

**Comment by sushmanthreddy** [comment]
- [interpolate] #21048

**Comment by sushmanthreddy** [comment]
- [flatten] #21049

**Comment by sushmanthreddy** [comment]
[adaptive_maxpoole_2d]#21051

**Comment by sushmanthreddy** [comment]
- [dropout3d] #21050

**Comment by guodh321** [comment]
> [log_softmax] #21254

Hi, could you help link the issue, thanks. @Ishticode

**Comment by yashj8** [comment]
-[] #21274

**Comment by yashj8** [comment]
-[] #21276

**Comment by curiousHG** [comment]
- [ log_softmax ] #21254

**Comment by curiousHG** [comment]
> > [log_softmax] #21254
> 
> Hi, could you help link the issue, thanks. @Ishticode

Done

**Comment by savita921** [comment]
- [ ] #21361

**Comment by sushmanthreddy** [comment]
- [avg_pool2d]  #21409

**Comment by sushmanthreddy** [comment]
- [bias_add] #21483

**Comment by nicolasb0** [comment]
[max_pool3d] #21513

**Comment by sushmanthreddy** [comment]
- [smooth_l1_loss] #21523

**Comment by NiteshK84** [comment]
- [ ] #21551
